packages/core/src/RenderingEngine/WSIViewport.ts

```diff
@@ -0,0 +1,533 @@
+   * for a URL parameter.
+   *
+   * @returns an imageID for video
+   */
+  public getCurrentImageId() {
+    return this.imageIds[0];
+  }
+
+  public getFrameNumber() {
+    return 1;
+  }
+
+  public getCamera(): ICamera {
+    const { parallelScale } = this.videoCamera;
+
+    const canvasCenter: Point2 = [
+      this.element.clientWidth / 2,
+      this.element.clientHeight / 2,
+    ];
+
+    // All other viewports have the focal point in canvas coordinates in the center
+    // of the canvas, so to make tools work the same, we need to do the same here
+    // and convert to the world coordinate system since focal point is in world coordinates.
+    const canvasCenterWorld = this.canvasToWorld(canvasCenter);
+
+    return {
+      parallelProjection: true,
+      focalPoint: canvasCenterWorld,
+      position: [0, 0, 0],
+      viewUp: [0, -1, 0],
+      parallelScale: this.element.clientHeight / 2 / parallelScale, // Reverse zoom direction back
+      viewPlaneNormal: [0, 0, 1],
+    };
+  }
+
+  public resetCamera = (): boolean => {
+    return true;
+  };
+
+  public getNumberOfSlices = (): number => {
+    return 1;
+  };
+
+  public getFrameOfReferenceUID = (): string => {
+    return this.frameOfReferenceUID;
+  };
+
+  public resize = (): void => {
+    const canvas = this.canvas;
+    const { clientWidth, clientHeight } = canvas;
+
+    // Set the canvas to be same resolution as the client.
+    if (canvas.width !== clientWidth || canvas.height !== clientHeight) {
+      canvas.width = clientWidth;
+      canvas.height = clientHeight;
+    }
+
+    this.refreshRenderValues();
+  };
+
+  /**
+   * Converts a VideoViewport canvas coordinate to a video coordinate.
+   *
+   * @param canvasPos - to convert to world
+   * @returns World position
+   */
+  public canvasToWorld = (canvasPos: Point2): Point3 => {
+    const pan: Point2 = this.videoCamera.panWorld; // In world coordinates
+    const worldToCanvasRatio: number = this.getWorldToCanvasRatio();
+
+    const panOffsetCanvas: Point2 = [
+      pan[0] * worldToCanvasRatio,
+      pan[1] * worldToCanvasRatio,
+    ];
+
+    const subCanvasPos: Point2 = [
+      canvasPos[0] - panOffsetCanvas[0],
+      canvasPos[1] - panOffsetCanvas[1],
+    ];
+
+    const worldPos: Point3 = [
+      subCanvasPos[0] / worldToCanvasRatio,
+      subCanvasPos[1] / worldToCanvasRatio,
+      0,
+    ];
+
+    return worldPos;
+  };
+
+  public async setWSI(imageIds: string[], client) {
+    console.log('Setting image ids', imageIds.length);
+    this.microscopyElement.style.background = 'red';
+    this.microscopyElement.innerText = 'Loading';
+    this.imageIds = imageIds;
+    const DicomMicroscopyViewer = await import('dicom-microscopy-viewer');
+    console.log('Loaded DICOMMicroscopyViewer', DicomMicroscopyViewer);
+    this.frameOfReferenceUID = null;
+    const metadata = this.imageIds.map((imageId) => {
+      const imageMetadata = client.getDICOMwebMetadata(imageId);
```

Comment:
we prob should go from metadata manager get eventually

Reply:
Agreed, but this is the required format right now internally, so getting it direct.  That avoids refetching the data, which makes things noticeably faster on first render.

Reply:
It is a lot of work to use dicom metadata manager.